### ISSUE TITLE
Hover Popover is not navigable via keyboard

### ISSUE THREAD

**Issue by sethrkeim** [palantir/blueprint#7661]
#### Environment

-   **Package version(s)**: @blueprintjs/core: 6.1.0
-   **Operating System**: MacOS Sequoia: 15.7.1
-   **Browser name and version**: Island: 1.79.21

#### Code Sandbox

Link to a minimal repro: https://ckwg88.csb.app/

#### Steps to reproduce

1. Navigate to the broken hover popover button using tab key to move element by element.
1. Attempt to navigate past the button by pressing the tab key again.
1. Focus is reset to the top of the page

#### Actual behavior

Focus is set to the top of the page.

#### Expected behavior

Focus should be captured in the popover, using the focus trap. When the `esc` key is pressed, the popover should close, and pressing the tab key should navigate focus to the next element on the page. 

#### Possible solution

When the `interactionKind` of an `Overlay2` is `hover`, the overlay should stay open when focus is on the start and end focus trap elements. In addition, pressing `esc` should hide the popover, so that it's possible to navigate past the popover. Alternatively, the focus trap could be abandoned for hover popovers altogether, and it could require arrow keys to navigate into the hover popover.

**Comment by vnkommu** [comment]
Would like to mega +1 this. This causes every Hover Popover to obstruct keyboard navigation:
1. Its contents are not navigable via keyboard tabbing
2. It stops the user from moving beyond the target of the Hover Popover (to the next interactive element) via the keyboard, bc the focus is reset to the top of the page
